src/serac/physics/utilities/functional/functional.hpp

```diff
@@ -357,20 +388,246 @@ class Functional<test(trial), execution_policy> : public mfem::Operator {
    * @brief Lightweight shim for mfem::Operator that produces the gradient of a @p Functional from a @p Mult
    */
   class Gradient : public mfem::Operator {
+
+    template < typename T >
+    struct Array3D{
+      Array3D() = default;
+      Array3D(int n1, int n2, int n3) : strides{n2 * n3, n3, 1}, data(n1 * n2 * n3) {}
+      auto & operator()(int i, int j, int k) { return data[i * strides[0] + j * strides[1] + k * strides[2]]; }
+      int strides[3];
+      std::vector < T > data;
+    };
```

Comment:
these are temporary containers being used until we have access to `axom::array`

Reply:
just FYI - I think what's currently in axom develop is sufficient to replace this

Reply:
I think this can be a follow on PR. I'll make an issue.